moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -561,11 +569,23 @@ int IKFastKinematicsPlugin::solve(KDL::Frame& pose_frame, const std::vector<doub
     case IKP_Lookat3D:
     case IKP_TranslationXY2D:
     case IKP_TranslationXYOrientation3D:
+      ROS_ERROR_NAMED(name_, "IK for this IkParameterizationType not implemented yet.");
+      return 0;
+
     case IKP_TranslationXAxisAngleZNorm4D:
+      pose_frame.M.GetRPY(roll, pitch, yaw);
+      ComputeIk(trans, &yaw, vfree.size() > 0 ? &vfree[0] : NULL, solutions);
+      return solutions.GetNumSolutions();
+
     case IKP_TranslationYAxisAngleXNorm4D:
+      pose_frame.M.GetRPY(roll, pitch, yaw);
+      ComputeIk(trans, &roll, vfree.size() > 0 ? &vfree[0] : NULL, solutions);
+      return solutions.GetNumSolutions();
+
     case IKP_TranslationZAxisAngleYNorm4D:
-      ROS_ERROR_NAMED(name_, "IK for this IkParameterizationType not implemented yet.");
-      return 0;
+      pose_frame.M.GetRPY(roll, pitch, yaw);
+      ComputeIk(trans, &pitch, vfree.size() > 0 ? &vfree[0] : NULL, solutions);
+      return solutions.GetNumSolutions();
 
     default:
       ROS_ERROR_NAMED(name_, "Unknown IkParameterizationType! Was the solver generated with an incompatible version "
```

Comment:
General comment: Could you modify your code again and verify that it works with this PR + `TranslationXAxisAngleZNorm4D`, as suggested in https://github.com/rdiankov/openrave/issues/596#issuecomment-403673192 ?

Reply:
Thank you for comment, Actually I do not have robot model to test `Translation$AxisAngle4D` so I think this PR should only deal with 'Translation$AxisAngle"#orm4D` and work for `Translation$AxisAngle4D`  when we found URDF to test.
I have confirmed with this new PR, see https://github.com/k-okada/SCORE/tree/master/src/scara_ikfast_plugin and https://github.com/ros-planning/moveit/issues/548#issuecomment-403674036

Reply:
I agree.